src/main/java/View/SquareView.java

```diff
@@ -57,24 +57,61 @@ public void setCircle(){
         circle.setOpacity(0.5);
     }
 
-    //get's the correct image from the resouces. add's the Image to the ImageView
-    //removes any previous imageview (incase it's a capture) and then add the imageview to the group
-    public void addPiece(Piece piece){
-        Image image = new Image(getPNGString(piece));
+    //Update the FENPiece and then get the correct image from the resouces and add/remove the imageview from the Group.
+    public void addPiece(Character charactor){
+        //check if the charactor is the same. If it is then return
+        if(charactor == this.FENPiece){
+            return;
+        }
+        //check if there was no piece previous on the spot. If so then add imageview to the Group
+        if (this.FENPiece == 'x'){
+            getChildren().add(imageview);
+        }
+        this.FENPiece = charactor;
+        //if there is no longer a piece on this square, remove the imageview and return
+        if (charactor == 'x'){
+            getChildren().remove(imageview);
+            return;
+        }
+        boolean isWhite = true;
+        if(Character.isLowerCase(charactor)){
+            isWhite = false;
+        }
+
+        Character lowerCase = Character.toLowerCase(charactor);
+        Image image = new Image(getPNGString(lowerCase, isWhite));
         this.imageview.setImage(image);
-        getChildren().remove(imageview);
-        getChildren().add(imageview);
     }
 
-    //remove a piece from this SquareView
-    public void removePiece(){
-        getChildren().remove(this.imageview);
-    }
 
     //returns the correct url (String) that the piece requires
-    public String getPNGString(Piece piece){
-        String output = piece.getPieceType().toString() + "-";
-        if(piece.getIsWhite()){
+    public String getPNGString(Character character, boolean isWhite){
+        String output = "";
+        switch (character){
+            case 'k':
+                output+="King";
+                break;
+            case 'q':
+                output+="Queen";
+                break;
+            case 'r':
+                output+="Rook";
+                break;
+            case 'n':
+                output+="Knight";
+                break;
+            case 'b':
+                output+="Bishop";
+                break;
+            case 'p':
+                output+="Pawn";
+                break;
```

Comment:
warning - no `default:` here, common source of bugs

Reply:
What should I do here? I've coded it so it should never get to a <default>. But if it ever gets to <default> what should I put there? No matter what I put it'll there will be an error later in the code (or display the wrong thing) so should I still have a <default>?

Reply:
at minimum just throw an error then:
```java
default:
    throw new IllegalStateException("invalid character found: " + character);
```
You can do similar for many other switch statements. I know what you mean that it's not actually necessary, and _technically_ it should be impossible to reach this scenario. But never rely on the impossible not happening...

Reply:
if you want to avoid the `default` altogether, then you can instead create an enum or something (or maybe even use `PieceType` to make the switch case exhaustive, and guarantee it will be one of the values. I personally prefer this but that's a bigger fix which isn't so necessary, but throwing exception will work for now.

Reply:
Going to leave as is for the moment so can merge and move on. As you said it's technically impossible currently so won't cause any issues in the short term. Can discuss the best way to solve this later